File: university-service/src/main/java/ua/com/foxminded/courseproject/service/PersonService.java
package ua.com.foxminded.courseproject.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import ua.com.foxminded.courseproject.dto.PersonDto;

import java.util.UUID;

public interface PersonService<T extends PersonDto> {
    public T findById(UUID id);

    public Page<T> findAll(Pageable pageable);

    public T save(T dto);

    public void delete(UUID id);

    public Boolean personExists(T personDto);

}

File: university-service/src/main/java/ua/com/foxminded/courseproject/service/PersonService.kt
package ua.com.foxminded.courseproject.service

import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import ua.com.foxminded.courseproject.dto.PersonDto
import java.util.*

interface PersonService<T : PersonDto> {
    fun findById(id: UUID): Mono<T>
    fun findAll(): Flux<T>
    fun save(dto: T): Mono<T>
    fun delete(id: UUID)
    fun personExists(personDto: T): Mono<Boolean>
}